### PR TITLE
'com.fasterxml.jackson.core' library upgrades

### PR DESCRIPTION
**Pull request by Gogetter**
### JIRA link (if applicable) ###
FPLA-814

### Change description ###
Override 'com.fasterxml.jackson.core' library version. Now using latest (2.10.0)

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/OtherProceedingsControllerTest.java [hunk 1]
```diff
@@ -75,31 +75,32 @@ void shouldReturnNoErrorsWhenPOnGoingProceedingIsDontKnow() throws Exception {
...
-                .data(ImmutableMap.<String, Object>builder().put("proceeding", map).build())
-                .build())
-            .build();
+                                                 .caseDetails(CaseDetails.builder()
+                                                                         .id(12345L)
+                                                                         .data(
+                                                                             ImmutableMap.<String, Object>builder().put(
+                                                                                 "proceeding", map).build())
+                                                                         .build())
+                                                 .build();
```
Comment (joez-f) [inline service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/OtherProceedingsControllerTest.java]: What happened with the indents here? 😄

Comment (lewisbirks) [reply service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/OtherProceedingsControllerTest.java]: Think that was when the checkstyle didn't import correctly on my machine, will fix

Comment (Gogetter) [reply service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/OtherProceedingsControllerTest.java]: fixed

File: service/build.gradle [hunk 2]
```diff
@@ -209,6 +212,11 @@ dependencies {
 
   compile group: 'javax.xml.bind', name: 'jaxb-api', version: '2.3.1'
 
+  //overriding com.fasterxml.jackson.core:jackson-databind to fix CVE issues suppressed
+  compile group: 'com.fasterxml.jackson.core', name: 'jackson-core', version: '2.10.0'
```
Comment (geek-soft) [inline service/build.gradle]: I'd be carefully using both Jackson 1 and Jackson 2. They might not be compatible.

Above comment assumes that spring boot brings other jackson modules in version 1.x.

File: service/build.gradle [hunk 3]
```diff
@@ -6,6 +6,7 @@ plugins {
   id 'org.springframework.boot' version '2.1.9.RELEASE'
   id 'com.github.ben-manes.versions' version '0.25.0'
   id 'org.sonarqube' version '2.8'
+  id 'project-report'
```
Comment (geek-soft) [inline service/build.gradle]: I am not convinced we need that. There are already command line tasks that provides same information just printed  to the terminal. For example to see dependencies tree `dependencies` task is already available.

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/HearingBookingDetailsControllerTest.java [hunk 4]
```diff
@@ -93,17 +92,19 @@ private HearingBooking createHearing(LocalDate hearingDate) {
     }
 
     private AboutToStartOrSubmitCallbackResponse makeRequest(HearingBooking hearingDetail) throws Exception {
-        HashMap<String, Object> map = objectMapper.readValue(objectMapper.writeValueAsString(hearingDetail),
-            new TypeReference<Map<String, Object>>() {
+        Map<String, Object> map = objectMapper.readValue(objectMapper.writeValueAsString(hearingDetail),
```
Comment (geek-soft) [inline service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/HearingBookingDetailsControllerTest.java]: Good change although I suggest another PR. It has nothing to do with Jackson upgrade.

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/OtherProceedingsControllerTest.java [hunk 5]
```diff
@@ -79,9 +79,8 @@ private Proceeding createProceeding(String onGoingProceeding) {
     }
 
     private AboutToStartOrSubmitCallbackResponse makeRequest(Proceeding proceeding) throws Exception {
-        HashMap<String, Object> map = MAPPER.readValue(MAPPER.writeValueAsString(proceeding),
-            new TypeReference<Map<String, Object>>() {
-            });
+        Map<String, Object> map = MAPPER.readValue(MAPPER.writeValueAsString(proceeding), new TypeReference<>() {
```
Comment (geek-soft) [inline service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/OtherProceedingsControllerTest.java]: Same here

File: service/src/main/java/uk/gov/hmcts/reform/fpl/model/RespondentParty.java [hunk 6]
```diff
@@ -1,6 +1,8 @@
 package uk.gov.hmcts.reform.fpl.model;
 
 import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
```
Comment (geek-soft) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/model/RespondentParty.java]: The are changes in this file needed?

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/HearingBookingDetailsControllerTest.java (hunk 4 above)
Comment (lewisbirks) [reply service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/HearingBookingDetailsControllerTest.java]: This was fixing issues that then occurred due to the upgrade

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/OtherProceedingsControllerTest.java (hunk 5 above)
Comment (lewisbirks) [reply service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/OtherProceedingsControllerTest.java]: same as above

File: service/src/main/java/uk/gov/hmcts/reform/fpl/model/RespondentParty.java (hunk 6 above)
Comment (Gogetter) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/model/RespondentParty.java]: yes the change was needed as jackson had issues (after version upgrade) deserialising given the custom `@Builder` constructor in this class.

File: service/build.gradle (hunk 3 above)
Comment (Gogetter) [reply service/build.gradle]: yes you are right, adding this was just a nice to have as we can have the html 😄

**Pull request merged**